src/strategies/curve/strategy-curve-stecrv.sol

```diff
@@ -0,0 +1,167 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.6.2;
+
+import "../../lib/erc20.sol";
+import "../../lib/safe-math.sol";
+
+import "../../interfaces/jar.sol";
+import "../../interfaces/steth.sol";
+import "../../interfaces/weth.sol";
+import "../../interfaces/curve.sol";
+import "../../interfaces/uniswapv2.sol";
+import "../../interfaces/controller.sol";
+
+import "../strategy-base.sol";
+
+interface ICurveFi {
+    function add_liquidity(
+        // stETH pool
+        uint256[2] calldata amounts,
+        uint256 min_mint_amount
+    ) external payable;
+
+    function balances(int128) external view returns (uint256);
+}
+
+contract StrategyCurveSteCRV is StrategyBase {
+    // Curve
+    IStEth public stEth = IStEth(0xae7ab96520DE3A18E5e111B5EaAb095312D7fE84); // lido stEth
+    IERC20 public steCRV = IERC20(0x06325440D014e39736583c165C2963BA99fAf14E); // ETH-stETH curve lp
+
+    // curve dao
+    ICurveGauge public gauge = ICurveGauge(0x182B723a58739a9c974cFDB385ceaDb237453c28); // stEthGauge
+    ICurveFi public curve = ICurveFi(0xDC24316b9AE028F1497c275EB9192a3Ea0f67022); // stEthSwap
+    address public mintr = 0xd061D61a4d941c39E5453435B6345Dc261C2fcE0;
+
+    // tokens we're farming
+    IERC20 public constant crv = IERC20(0xD533a949740bb3306d119CC777fa900bA034cd52);
+    IERC20 public constant ldo = IERC20(0x5A98FcBEA516Cf06857215779Fd812CA3beF1B32);
+
+    // How much CRV tokens to keep
+    uint256 public keepCRV = 500;
+    uint256 public keepCRVMax = 10000;
+
+    constructor(
+        address _governance,
+        address _strategist,
+        address _controller,
+        address _timelock
+    )
+        public
+        StrategyBase(address(steCRV), _governance, _strategist, _controller, _timelock)
+    {
+        steCRV.approve(address(gauge), uint256(-1));
+        stEth.approve(address(curve), uint256(-1));
+        ldo.safeApprove(address(univ2Router2), uint256(-1));
+        crv.approve(address(univ2Router2), uint256(-1));
+    }
+
+    // swap for eth
+    receive() external payable {}
+
+    // **** Getters ****
+
+    function balanceOfPool() public override view returns (uint256) {
+        return gauge.balanceOf(address(this));
+    }
+
+    function getName() external override pure returns (string memory) {
+        return "StrategyCurveStETH";
+    }
+
+    function getHarvestable() external view returns (uint256) {
+        return gauge.claimable_reward(address(this), address(crv));
+    }
+
+    function getHarvestableEth() external view returns (uint256) {
+        uint256 claimableCrv = gauge.claimable_reward(address(this), address(ldo));
```

Comment:
you swapped the variable names with the addresses

Reply:
Not sure if I get it right.. Do you mean it should be `gauge.claimable_reward(address(ldo), address(this));`?